src/connect/connect.cpp

```diff
@@ -139,7 +139,12 @@ SQLRETURN Connect::SetConnection() {
 	config.SetOptionsByName(config_map);
 
 	bool cache_instance = database != IN_MEMORY_PATH;
-	dbc->env->db = instance_cache.GetOrCreateInstance(database, config, cache_instance);
+
+    try {
+	    dbc->env->db = instance_cache.GetOrCreateInstance(database, config, cache_instance);
+    } catch (std::exception &ex) {
+        return SetDiagnosticRecord(dbc, SQL_ERROR, "SQLDriverConnect", ex.what(), SQLStateType::ST_IM003, "");
```

Comment:
That looks right to me!
We usually use `ErrorData error(ex);` in these places
We could decide to only throw this error on IOException and still propagate the error for other types?

But I'm not sure if that makes sense in this context, this should do the job 👍